# Rhino Africa Inquiry Data Analysis
**Student Name:** [Your Name]

**Student ID:** [Your ID]

**Course:** ISM4641 - Fall 2025

---

## Assignment Overview
This notebook analyzes customer inquiry data from Rhino Africa (`inquiries_mini.csv`) using custom Python functions created in the `my_rhino_lastname.py` module. The analysis covers geographic patterns, budget trends, travel timing, and customer segmentation.

## Section 1: Data Loading and Overview (8 points)

In this section, we'll import our custom module and load the inquiry data to get a basic understanding of the dataset.

In [69]:
# Import your custom module
import my_rhino_palani as my_rhino # change my_rhino_lastname to be the exact file name you are using for your file

# Load the inquiries_mini.csv data using your function
inquiries = my_rhino.load_inquiries_data('inquiries_mini.csv')

# Display basic information about the dataset
my_rhino.print_data_summary(inquiries)

Total number of inquiries: 45203
Number of columns: 16
Column names:dict_keys(['inquiry_id', 'customer_id', 'timestamp_utc', 'timestamp_local', 'timezone_offset', 'completed', 'customer_country', 'customer_city', 'customer_latitude', 'customer_longitude', 'travel_year', 'travel_month', 'num_adults', 'num_children', 'budget_currency', 'budget_value'])


### Data Overview Commentary
This imports the my_rhino_palani python file into this jupyter notebook as a module. This loads the data from inquiries_mini.csv into a variable called inquiries. This prints a summary of the data in the inquiries_mini.csv. It gives us the number of inquiries, number of columns, and column names in the inquiries_mini excel file.

## Section 2: Geographic Analysis (8 points)

Let's analyze where our customers are coming from by examining inquiry patterns by country.

In [70]:
# Analyze inquiries from specific countries
us_inquiries = my_rhino.count_inquiries_by_country(inquiries, 'United States')
uk_inquiries = my_rhino.count_inquiries_by_country(inquiries, 'United Kingdom')
sa_inquiries = my_rhino.count_inquiries_by_country(inquiries, 'South Africa')

print(f"United States inquiries: {us_inquiries}")
print(f"United Kingdom inquiries: {uk_inquiries}")
print(f"South Africa inquiries: {sa_inquiries}")

Found 14221 inquiries from United States.
Found 9069 inquiries from United Kingdom.
Found 8995 inquiries from South Africa.
United States inquiries: None
United Kingdom inquiries: None
South Africa inquiries: None


In [71]:
# Create a list of top countries by manually checking several countries
# You'll need to test different countries and create your top 5 list
countries_to_check = ['United States', 'United Kingdom', 'South Africa', 'Canada', 'Australia', 'France', 'Belgium']
country_counts = []

for country in countries_to_check:
    count = my_rhino.count_inquiries_by_country(inquiries, country)
#if count > 0:
country_counts.append((country, count))

# Sort by count (descending)
country_counts.sort(key=lambda x: x[1], reverse=True)

print("Top Countries by Inquiry Count:")
for i, (country, count) in enumerate(country_counts[:5], 1):
    print(f"{i}. {country}: {count} inquiries")

Found 14221 inquiries from United States.
Found 9069 inquiries from United Kingdom.
Found 8995 inquiries from South Africa.
Found 2218 inquiries from Canada.
Found 2566 inquiries from Australia.
Found 2107 inquiries from France.
Found 331 inquiries from Belgium.
Top Countries by Inquiry Count:
1. Belgium: None inquiries


### Geographic Patterns Commentary

 The top five countries with the most inquiries are United States, United Kingdom, South Africa, Canada, Australia in descending order. There is a significant decrease in inquiries between Top 3 country and Top 4 country. It is surprising that the difference in inquiries between Top 1 country and Top 5 country is greater than 10,000.

## Section 3: Budget Analysis (8 points)

Now let's examine the budget information from customer inquiries to understand spending patterns.

In [72]:
# Get budget statistics
budget_stats = my_rhino.get_budget_statistics(inquiries)

print("Budget Statistics:")
print(f"Total Budget Value: ${budget_stats['total']:,.2f}")
print(f"Average Budget: ${budget_stats['average']:,.2f}")
print(f"Minimum Budget: ${budget_stats['min']:,.2f}")
print(f"Maximum Budget: ${budget_stats['max']:,.2f}")
print(f"Number of Valid Budgets: {budget_stats['count']}")

RecursionError: maximum recursion depth exceeded

In [73]:
# Calculate percentage of inquiries with valid budget information
total_inquiries = len(inquiries)
valid_budgets = budget_stats['count']
budget_completion_rate = (valid_budgets / total_inquiries) * 100

print(f"\nBudget Completion Analysis:")
print(f"Total Inquiries: {total_inquiries}")
print(f"Inquiries with Valid Budgets: {valid_budgets}")
print(f"Budget Completion Rate: {budget_completion_rate:.1f}%")

NameError: name 'budget_stats' is not defined

### Budget Patterns Commentary


The average budget is $16182. The average budget tells us that the budget can vary from anywhere in between the minimum budget of $4000 and the maximum budget of $4000. Some insights are that the travel agency is a well-known company due the count of budgets being 45203.

## Section 4: Travel Timing Patterns (8 points)

Let's analyze when customers prefer to travel by examining the travel month data.

In [74]:
# Get popular travel months
popular_months = my_rhino.find_popular_travel_months(inquiries)

print("Popular Travel Months (Ranked by Frequency):")
for i, (month, count) in enumerate(popular_months, 1):
    print(f"{i}. {month}: {count} inquiries")

NameError: name 'month' is not defined

In [75]:
# Additional analysis: peak vs off-peak seasons
if popular_months:
    peak_month = popular_months[0]
    print(f"\nSeasonal Analysis:")
    print(f"Peak Month: {peak_month[0]} with {peak_month[1]} inquiries")

    # Calculate what percentage of travel months are specified
    total_with_months = sum(count for month, count in popular_months)
    total_inquiries = len(inquiries)
    month_completion_rate = (total_with_months / total_inquiries) * 100
    print(f"Travel Month Completion Rate: {month_completion_rate:.1f}%")

NameError: name 'popular_months' is not defined

### Travel Timing Commentary

 The most popular months are June, July, and August and it aligns with typical safari seasons due to the summer season. The count of inquiries of these months are over 4,000. The count of inquiries increases from March till August, the spring/summer season, and starts decreasing from September till February, the fall/winter season.

## Section 5: Customer Segmentation (8 points)

Finally, let's analyze customer behavior patterns by looking at completion rates and group sizes.

In [76]:
# Display completion report
print("Inquiry Completion Analysis:")
my_rhino.print_completion_reports(inquiries)

Inquiry Completion Analysis:


0

In [77]:
# Analyze group sizes
group_categories = my_rhino.categorize_group_sizes(inquiries)

print("\nGroup Size Analysis:")
total_categorized = sum(group_categories.values())

for category, count in group_categories.items():
    percentage = (count / total_categorized) * 100
    print(f"{category}: {count} inquiries ({percentage:.1f}%)")

print(f"\nTotal Categorized Inquiries: {total_categorized}")

NameError: name 'row' is not defined

In [68]:
# Additional insight: Most common customer type
most_common_type = max(group_categories, key=group_categories.get)
print(f"\nCustomer Insights:")
print(f"Most Common Customer Type: {most_common_type} ({group_categories[most_common_type]} inquiries)")

# Calculate data completeness for group size information
total_inquiries = len(inquiries)
group_data_rate = (total_categorized / total_inquiries) * 100
print(f"Group Size Data Completion Rate: {group_data_rate:.1f}%")

NameError: name 'group_categories' is not defined

### Customer Behavior Commentary


There are 23,713 completed inquiries and 21,490 incompleted inquiries so the completion rate is higher than the incompletion rate. The Couple groap size is the most common. This information can tell Rhino Africa that it should focus on marketing for couples and maintain its operations to about to execute about half of its number of inquiries.

## Summary and Conclusions


The most important findings about Rhino Africa's customer inquiries are that the highest client base consists of clients from the United States, clients that are couples, and in the peak season of Summer. A recommendation is for Rhino Africa to increase advertising targeting couples in the United States during the Summer to increase its client base.
